crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -63,10 +63,14 @@ impl TestCallArg {
 
     async fn call_arg_from_id(object_id: ObjectID, state: &AuthorityState) -> ObjectArg {
         let object = state.get_object(&object_id).await.unwrap().unwrap();
-        if object.is_shared() {
-            ObjectArg::SharedObject(object_id)
-        } else {
-            ObjectArg::ImmOrOwnedObject(object.compute_object_reference())
+        match &object.owner {
+            Owner::AddressOwner(_) | Owner::ObjectOwner(_) | Owner::Immutable => {
+                ObjectArg::ImmOrOwnedObject(object.compute_object_reference())
+            }
+            Owner::Shared { shared_at_version } => ObjectArg::SharedObject {
+                id: object_id,
+                shared_at_version: *shared_at_version,
+            },
```

Comment:
we need a test somewhere in this file verifying that a validator will refuse to sign a transaction with a bad shared_at_version.

Reply:
Is that not what `crates/sui/tests/shared_objects_version_tests.rs` is for?

Reply:
my mistake, missed that one.